x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceResponseHandlerTests.java

```diff
@@ -32,62 +33,77 @@
-        MatcherAssert.assertThat(((ElasticsearchStatusException) exception.getCause()).status(), is(RestStatus.METHOD_NOT_ALLOWED));
-    }
+    private final FailureTestCase failureTestCase;
 
-    public void testCheckForFailureStatusCode_ThrowsFor413() {
-        var exception = expectThrows(ContentTooLargeException.class, () -> callCheckForFailureStatusCode(413, "id"));
-        assertTrue(exception.shouldRetry());
-        MatcherAssert.assertThat(
-            exception.getCause().getMessage(),
-            containsString("Received a content too large status code for request from inference entity id [id] status [413]")
-        );
-        MatcherAssert.assertThat(((ElasticsearchStatusException) exception.getCause()).status(), is(RestStatus.REQUEST_ENTITY_TOO_LARGE));
+    public ElasticInferenceServiceResponseHandlerTests(FailureTestCase failureTestCase) {
+        this.failureTestCase = failureTestCase;
     }
 
-    public void testCheckForFailureStatusCode_ThrowsFor500_WithShouldRetryTrue() {
-        var exception = expectThrows(RetryException.class, () -> callCheckForFailureStatusCode(500, "id"));
-        assertTrue(exception.shouldRetry());
-        MatcherAssert.assertThat(
-            exception.getCause().getMessage(),
-            containsString("Received a server error status code for request from inference entity id [id] status [500]")
+    @ParametersFactory
+    public static Iterable<FailureTestCase[]> parameters() throws Exception {
+        return java.util.Arrays.asList(
+            new FailureTestCase[][] {
+                {
+                    new FailureTestCase(
+                        400,
+                        RestStatus.BAD_REQUEST,
+                        "Received a bad request status code for request from inference entity id [id] status [400]",
+                        false
+                    ) },
+                {
+                    new FailureTestCase(
+                        402,
+                        RestStatus.PAYMENT_REQUIRED,
+                        "Received an unsuccessful status code for request from inference entity id [id] status [402]",
+                        false
+                    ) },
+                {
+                    new FailureTestCase(
+                        405,
+                        RestStatus.METHOD_NOT_ALLOWED,
+                        "Received a method not allowed status code for request from inference entity id [id] status [405]",
+                        false
+                    ) },
+                {
+                    new FailureTestCase(
+                        413,
+                        RestStatus.REQUEST_ENTITY_TOO_LARGE,
+                        "Received a content too large status code for request from inference entity id [id] status [413]",
+                        true
+                    ) },
+                {
+                    new FailureTestCase(
+                        500,
+                        RestStatus.BAD_REQUEST,
```

Comment:
Just curious: This means, that we translate the server error to a client error for the ES client, right? Do we do that to avoid the client also retrying?

Reply:
No it's more so that our alerting doesn't trigger on the 500s. Our internal alerting looks for 5xx errors I believe. We have an issue to address that so we can forward the response on but haven't gotten to it yet.